wicket-examples/src/main/java/org/apache/wicket/examples/csp/RfcPage.java

```diff
@@ -0,0 +1,180 @@
+	public RfcPage()
+	{
+		final Label c1 = new Label("c1", LambdaModel.of(this::getCounter1)) {
+			@Override
+			public void renderHead(IHeaderResponse response)
+			{
+				super.renderHead(response);
+				// render random header item
+				if (Math.random() > 0.5)
+				{
+					response.render(OnDomReadyHeaderItem.forScript("otherGlobalFunction('c1 rendered more')"));
+				} else {
+					response.render(OnDomReadyHeaderItem.forScript("otherGlobalFunction('c1 rendered less')"));
+				}
+			}
+		};
+		c1.setVisible(false);
+		c1.setOutputMarkupId(true);
+		add(c1);
+		final Label c2 = new Label("c2", LambdaModel.of(this::getCounter2));
+		c2.setOutputMarkupId(true);
+		add(c2);
+		add(new AjaxLink<Void>("c1-link")
+		{
+			@Override
+			protected void onComponentTag(ComponentTag tag)
+			{
+				super.onComponentTag(tag);
+				if (isEnabledInHierarchy()) {
+					// XXX this is another issue which should be fixed globally, href="javascript:;" should go
+					// XXX there's no need in no-op href if there's actually no link
+					if (!Strings.isEmpty(tag.getAttribute("href"))) {
+						tag.remove("href");
+					}
+				}
+			}
+
+			@Override
+			public void onClick(AjaxRequestTarget target)
+			{
+				counter1++;
+				c1.setVisible(true);
+				target.add(c1);
+				target.prependRemoteFunctionCall("testFunc1Pre", "something1", 1.5);
+				target.appendRemoteFunctionCall("testFunc1", "something2", 1);
+				JSONObject obj = new JSONObject();
+				obj.put("horse", "big");
+				obj.put("cat", "small");
+				target.appendRemoteFunctionCall("testFuncFunc", "myJson", obj);
+				target.appendJavaScript("otherGlobalFunction('appended javascript');");
+			}
+		});
+		add(new AjaxFallbackLink<Void>("c2-link")
+		{
+			@Override
+			public void onClick(Optional<AjaxRequestTarget> targetOptional)
+			{
+				counter2++;
+				targetOptional.ifPresent(target -> {
+					target.add(c2);
+					target.prependRemoteFunctionCall("testFunc2Pre", "1", "ABC");
+					target.appendRemoteFunctionCall("testFunc2", "1", Math.random());
+				});
+			}
+		});
+	}
+
+	@Override
+	public void renderHead(IHeaderResponse response)
+	{
+		response.render(OnDomReadyHeaderItem.forScript(
+				"otherGlobalFunction = function(fn, args) {" +
+						"console.log(fn, args);" +
+						"var cnt = document.getElementById('resultsContainer');" +
+						"cnt.innerHTML = fn + ':' + JSON.stringify(args) + \"\\n\" + cnt.innerHTML;" +
+						"};" +
+						"Wicket.Ajax.RFC.testFunc1 = function(){otherGlobalFunction('testFunc1', arguments)};" +
+						"Wicket.Ajax.RFC.testFuncFunc = function(){otherGlobalFunction('testFuncFunc', arguments)};" +
+						"Wicket.Ajax.RFC.testFunc1Pre = function(){otherGlobalFunction('testFunc1Pre', arguments)};" +
+						"Wicket.Ajax.RFC.testFunc2 = function(){otherGlobalFunction('testFunc2', arguments)};" +
+						"Wicket.Ajax.RFC.testFunc2Pre = function(){otherGlobalFunction('testFunc2Pre', arguments)};" +
+						"eval('alert(\"this popup should not be displayed\")');"
+		));
```

Comment:
Let's rework this into a proper `.js` file that sets up the RFC entries. Users will follow this terrible/inline example.